tests/playwright-test/config.spec.ts

```diff
@@ -72,6 +72,92 @@ test('should prioritize command line timeout over project timeout', async ({ run
   expect(result.output).toContain('Test timeout of 500ms exceeded.');
 });
 
+test('should default to failOnFlakyTests false', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'a.test.js': `
+      import { test, expect } from '@playwright/test';
+      test('flake', async ({}, testInfo) => {
+        expect(testInfo.retry).toBe(1);
+      });
+    `,
+  });
+  expect(result.exitCode).not.toBe(0);
+  expect(result.flaky).toBe(0);
+});
+
+test('should prioritize command line --fail-on-flaky-tests flag over config failOnFlakyTests', async ({ runInlineTest }) => {
+  const result = await runInlineTest({
+    'playwright.config.ts': `
+      module.exports = {
+        failOnFlakyTests: false
+      };
+    `,
+    'a.test.js': `
+      import { test, expect } from '@playwright/test';
+      test('flake', async ({}, testInfo) => {
+        expect(testInfo.retry).toBe(1);
+      });
+    `,
+  }, { 'retries': 1, 'fail-on-flaky-tests': true });
+  expect(result.exitCode).not.toBe(0);
+  expect(result.flaky).toBe(1);
+});
+
+test('should support failOnFlakyTests config option', async ({ runInlineTest }) => {
```

Comment:
this one test should be enough to cover this new flag, let's remove the rest.

Reply:
Should we not have two tests, one for the CLI option, and one for the config option?

Reply:
Right! I mean let's remove the rest you added. Let's not remove existing tests :D